### PR TITLE
Use watchdog for a more robust file watching for mesop/bin

### LINKED ISSUE #150: Support hot reloading from multiple files using `mesop` CLI (via pip install)
Use https://github.com/gorakhargosh/watchdog?

### PR DESCRIPTION
**Pull request by wwwillchen**
Closes #150.

Note: this doesn't work with Python relative imports which seems relatively less popular than Python absolute imports.

### CODE REVIEW

File: mesop/bin/bin.py [hunk 1]
```diff
@@ -84,37 +85,113 @@ def main(argv: Sequence[str]):
...
+
+  # Initially track all the files on the file system and then rely on watchdog.
+  for root, dirnames, files in os.walk(workspace_dir_path):
+    # Filter out directories starting with "." because they
+    # can be special directories like .venv directories which
+    # can have lots of Python files that are definitely not application Python modules.
+    dirnames[:] = [d for d in dirnames if not d.startswith(".")]
+
+    for file in files:
+      if file.endswith(".py"):
```
Comment (richard-to) [inline mesop/bin/bin.py:169]: [question] How does this handle the `__init__.py` case?

File: mesop/bin/bin.py [hunk 2]
```diff
@@ -84,37 +85,113 @@ def main(argv: Sequence[str]):
...
+    src_path = cast(str, event.src_path)
+    if src_path.endswith(".py"):
+      print(f"Stopped watching deleted Python module: {event.src_path}")
+      remove_app_module(
+        workspace_dir_path=self.workspace_dir_path,
+        app_module_path=event.src_path,
+      )
+
+
+def fs_watcher(absolute_path: str):
```
Comment (richard-to) [inline mesop/bin/bin.py:154]: [question] So absolute path is the full path to the "main.py" file?

**Review by richard-to — APPROVED** [review]
Looks good. I think I only had one question about the __init__.py case.

File: mesop/bin/bin.py (hunk 2 above)
Comment (wwwillchen) [reply mesop/bin/bin.py:154]: yup

File: mesop/bin/bin.py (hunk 1 above)
Comment (wwwillchen) [reply mesop/bin/bin.py:180]: great catch - I forgot to handle `__init__.py`

**Comment by wwwillchen** [comment]
Thanks for the review

**Pull request merged**